src/mg_biome.h

```diff
@@ -56,10 +66,122 @@ class Biome : public ObjDef, public NodeResolver {
 	virtual void resolveNodeNames();
 };
 
-class BiomeManager : public ObjDefManager {
+
+////
+//// BiomeGen
+////
+
+enum BiomeGenType {
+	BIOMEGEN_ORIGINAL,
+};
+
+struct BiomeParams {
+	virtual void readParams(const Settings *settings) = 0;
+	virtual void writeParams(Settings *settings) const = 0;
+	virtual ~BiomeParams() {}
+
+	int seed;
```

Comment:
I'd like to fix this in a commit outside of this PR instead, since it's not really related to the mapgen cleanup and it touches noise.cpp quite substantially.